src/Internal/Service/Blockfrost.purs

```diff
@@ -116,3 +151,105 @@ handleBlockfrostResponse (Right { status: Affjax.StatusCode statusCode, body })
+  pure $ handle404AsNothing $ unwrapBlockfrostDatum <$> handleBlockfrostResponse
+    response
+
+getScriptInfo
+  :: ScriptHash
+  -> BlockfrostServiceM (Either ClientError (Maybe BlockfrostScriptInfo))
+getScriptInfo scriptHash = do
+  response <- blockfrostGetRequest (GetScriptInfo scriptHash)
+  pure $ handle404AsNothing $ handleBlockfrostResponse response
+
+getScriptByHash
+  :: ScriptHash -> BlockfrostServiceM (Either ClientError (Maybe ScriptRef))
+getScriptByHash scriptHash = runExceptT $ runMaybeT do
+  info <- MaybeT $ ExceptT $ getScriptInfo scriptHash
+  cbor <- MaybeT $ ExceptT $ do
+    response <- blockfrostGetRequest (GetScriptCbor scriptHash)
+    pure $ handle404AsNothing $ unwrapBlockfrostCbor <$>
+      handleBlockfrostResponse response
+  let
+    script = lmap (ClientDecodeJsonError "Error decoding script") $
+      wrapScriptType (scriptInfoType info) cbor
+  MaybeT $ ExceptT $ pure $ Just <$> script
+  where
+  wrapScriptType scriptType x = case scriptType of
+    NativeScript -> NativeScriptRef <$> decodeNativeScript x
+    PlutusV1Script -> pure $ PlutusScriptRef $ plutusV1Script x
+    PlutusV2Script -> pure $ PlutusScriptRef $ plutusV2Script x
+
+--------------------------------------------------------------------------------
+-- `getDatumByHash` response parsing
+--------------------------------------------------------------------------------
+
+newtype BlockfrostDatum = BlockfrostDatum (Maybe Datum)
+
+derive instance Newtype BlockfrostDatum _
+
+unwrapBlockfrostDatum :: BlockfrostDatum -> Maybe Datum
+unwrapBlockfrostDatum = unwrap
+
+instance DecodeAeson BlockfrostDatum where
+  decodeAeson aeson
+    | isNull aeson = pure $ BlockfrostDatum Nothing
+    | otherwise = do
+        cbor <- aesonObject (flip getFieldOptional "cbor") aeson
+        pure $ BlockfrostDatum $ deserializeData =<< cbor
+
+--------------------------------------------------------------------------------
+-- `getScriptByHash` response parsing
+--------------------------------------------------------------------------------
+
+data BlockfrostScriptLanguage = NativeScript | PlutusV1Script | PlutusV2Script
+
+derive instance Generic BlockfrostScriptLanguage _
+
+instance Show BlockfrostScriptLanguage where
+  show = genericShow
+
+instance DecodeAeson BlockfrostScriptLanguage where
+  decodeAeson = aesonString $ case _ of
+    "native" -> pure NativeScript
+    "plutusV1" -> pure PlutusV1Script
+    "plutusV2" -> pure PlutusV2Script
+    invalid ->
+      Left $ TypeMismatch $
+        "language: expected 'native' or 'plutusV{1|2}', got: " <> invalid
+
+-- Do not parse fields other than `type`, cuz we do not need them yet
+data BlockfrostScriptInfo = BlockfrostScriptInfo
+  { type :: BlockfrostScriptLanguage }
+
+instance DecodeAeson BlockfrostScriptInfo where
+  decodeAeson aeson = do
+    mType <- aesonObject (flip getFieldOptional "type") aeson
+    type_ <- note (AtKey "type" MissingValue) mType
```

Comment:
`type` is a required field that cannot be `null`, so we can use `getField` instead of `getFieldOptional`:
```suggestion
    type_ <- aesonObject (flip getField "type") aeson
```